Mathlib/RingTheory/Adjoin/PowerBasis.lean

```diff
@@ -85,7 +85,7 @@ variable [IsScalarTower R S A] {B : PowerBasis S A}
 `IsIntegral R (B.basis.repr (B.gen ^ n) i)` for all `i` if
 `minpoly S B.gen = (minpoly R B.gen).map (algebraMap R S)`. This is the case if `R` is a GCD domain
 and `S` is its fraction ring. -/
-theorem repr_gen_pow_isIntegral (hB : IsIntegral R B.gen) [IsDomain S]
+theorem repr_gen_pow_isIntegral (hB : IsIntegral R B.gen) [Nontrivial S]
```

Comment:
At least in this lemma, you don't need `Nontrivial S`, because if `S = 0`, then `B.basis.repr (B.gen ^ n) i = 0`, so the goal is `IsIntegral R 0`.

Please add a lemma `@[nontriviality] IsIntegral.of_subsingleton_right [Subsingleton S] (x : S) : IsIntegral R x`, add a version assuming `[Subsingleton R]` and getting `Subsingleton S` from `MulActionWithZero.subsingleton`. Then `nontriviality S` after `intro i` will add the required `[Nontrivial S]` assumption.